site/docs/api/serialization.md

```diff
@@ -3,6 +3,87 @@ title: Serialization
+- [com.hazelcast.nio.serialization.StreamSerializer](https://docs.hazelcast.org/docs/latest/javadoc/com/hazelcast/nio/serialization/StreamSerializer.html) &
+  [com.hazelcast.nio.serialization.ByteArraySerializer](https://docs.hazelcast.org/docs/latest/javadoc/com/hazelcast/nio/serialization/ByteArraySerializer.html)
+
+The following table provides a comparison between them to help you in 
+deciding which interface to use in your applications.
+|      Serialization interface      |                                                                      Advantages                                                                      |                                               Drawbacks                                              |
+|:---------------------------------:|:----------------------------------------------------------------------------------------------------------------------------------------------------:|:----------------------------------------------------------------------------------------------------:|
+|            Serializable           | <ul><li>Easy to start with, requires no implementation</li></ul>                                                                                     | <ul><li>CPU intensive</li><li>Space inefficient</li></ul>                                            |
+|           Externalizable          | <ul><li>Faster and more space efficient than Serializable</li></ul>                                                                                  | <ul><li>CPU intensive</li><li>Space inefficient</li><li>Requires implementation</li></ul>            |
+|          DataSerializable         | <ul><li>Faster and more space efficient than java standard interfaces</li></ul>                                                                      | <ul><li>Requires implementation</li></ul>                                                            |
+|     IdentifiedDataSerializable    | <ul><li>Relatively fast and space efficient</li></ul>                                                                                                | <ul><li>Requires implementation</li><li>Requires factory registration during cluster setup</li></ul> |
+|              Portable             | <ul><li>Faster and more space efficient than java standard interfaces</li><li>Supports versioning</li><li>Supports partial deserialization</li></ul> | <ul><li>Requires implementation</li><li>Requires factory registration during cluster setup</li></ul> |
+| [Stream&#124;ByteArray]Serializer | <ul><li>The fastest and lightest out of supported interfaces</li></ul>                                                                               | <ul><li>Requires implementation</li><li>Requires registration during cluster setup</li></ul>         |
+
+Below you can find rough performance numbers one can expect when 
+employing each of those strategies. A straightforward benchmark which continuously serializes and then 
```

Comment:
Maybe a link to the benchmark source for reproducibility ?

Reply:
I'm not sure about this, I wanted the readers just to have an idea about the potential performance. Where would that sit if we wanted to add it?

Reply:
I think https://github.com/hazelcast/hazelcast-jmh is a good candidate for that. My thinking was they can take the benchmarking boilerplate and plug their own POJOs to see the real performance impact for their actual objects.

Reply:
Makes sense. I need to polish it then a bit and add it.